### DISCUSSION TITLE
Button vs Link for CancelButton

### DISCUSSION THREAD

**Discussion by josch87** [neuefische/questions#640] [Web] [answered]
Usually when linking to another page you are using an anchor-element or in Next.js the Link-Component. Next to my "Add new contact" button (a button element) which submits a form (user interaction) I have a [cancel button](https://github.com/josch87/Gunther/blob/94bd6f381ef0216f0e67f7f773cd538f043a11cb/components/ContactForm/ContactForm.js#L300-L305) which leads back to the previous visited page using [router.back()](https://github.com/josch87/Gunther/blob/94bd6f381ef0216f0e67f7f773cd538f043a11cb/components/ContactForm/ContactForm.js#L48-L51). 

```
  function handleCancelButton() {
    router.back();
  }
```

```
          <Button type="button" onClick={handleCancelButton}>
            Cancel
          </Button>
          <Button type="submit" buttonType="primary">
            Add new contact
          </Button>
```

1. Is this a use case where I should use a button instead of a link or would you recommend changing it to a link instead which is styled exactly like the "Add new contact" button? 
2. When using a link, how would you implement the handleCancelButton function (can you name this a Callback-Function?)? Because usually, a link uses an href instead. 
3. How would you decide when the handleCancelButton function has more logic to it such as checking if the user typed any data into the form and if the data is not NULL alerting the user if he/she really wants to cancel (and lose all typed data) before redirecting to the previous page? 

Thank you in advance for your advice.

**Comment by AliaksandraNovikava** [comment, marked as answer]
Hey @josch87 .
I found this article regarding your questions: [Set onClick on Link component in React](https://simplefrontend.com/set-onclick-on-link-component-in-react/#:~:text=We%20can%20set%20the%20onClick,for%20any%20other%20React%20element.&text=Use%20curly%20braces%20and%20set,execute%20the%20doSomething()%20function.)
I hope this can be a little bit helpful to you.

**Reply by josch87** [reply]
Thank you. Looks like they also recommend using a button and a handler that calls router. 

I also talked to Jessica and Dominik and they agree that I can use a button here instead of a link.
